src/audio/module_adapter/module/cadence.c

```diff
@@ -231,10 +362,11 @@ static int cadence_codec_init(struct processing_module *mod)
 
 	/* copy the setup config only for the first init */
 	if (codec->state == MODULE_DISABLED && codec->cfg.avail) {
-		struct module_config *setup_cfg = &cd->setup_cfg;
+		setup_cfg = &cd->setup_cfg;
 
 		/* allocate memory for set up config */
-		setup_cfg->data = rballoc(0, SOF_MEM_CAPS_RAM, codec->cfg.size);
+		setup_cfg->data = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+					  codec->cfg.size);
```

Comment:
same - it's overwritten. In general - usually `data` means some unstructured data and as such initialising it to 0 rarely makes sense.